### PR TITLE
Add `is_lazy_conjugate`

### PR DESCRIPTION
**Pull request by MasonProtter**
This adds a function `is_lazy_conjugate` that can tell if a given array is supposed to lazyily take complex conjugates or not. It's meant to understand array wrappers, but if two nested wrappers have the exact same type in a row it will get confused. Not sure if there's a way around that other than whack-a-mole.

`Adjoint` is the only lazy-`conj` wrapper I know of, but the idea is that if anyone else has such types they can register here.

### CODE REVIEW

**Review by chriselrod — COMMENTED** [review]
Some methods return `Static.True()` or `Static.False()`, which would have the advantage of letting you dispatch on whether it's true or false, and wouldn't require const prop to succeed at the user's side.

If you want to switch to `Static.True`/`Static.False`, it'd probably be worth using `@inferred` in all the tests, to make sure it succeeds.

**Comment by chriselrod** [comment]
The change to `StaticBool`s should be pretty trivial:
```julia
julia> using Static

julia> !True()
False()

julia> !False()
True()
```

File: src/ArrayInterface.jl [hunk 1]
```diff
@@ -841,6 +841,51 @@ end
...
+
+    julia> b = a'
+    1×2 adjoint(transpose(adjoint(::Vector{Complex{Int64}}))) with eltype Complex{Int64}:
+     1+1im  1-1im
+
+    julia> ArrayInterface.is_lazy_conjugate(b)
+    false
+
+"""
+is_lazy_conjugate(::T) where {T <: AbstractArray} = _is_lazy_conjugate(T, false)
```
Comment (chriselrod) [inline src/ArrayInterface.jl:869]: The change to `StaticBool`s should be pretty trivial:
```julia
julia> using Static

julia> !True()
False()

julia> !False()
True()
```
That is, I think replace `false` here with `False()` is all it'd take.

**Comment by MasonProtter** [comment]
Okay, I've switch to `True/False`

**Pull request merged**